docs/design/mono/diagnostics-tracing.md

```diff
@@ -454,6 +468,33 @@ won't get any additional events, just trigger a heap dump.
 
 Combining different sessions including different GC information opens up ability to track all GC allocations during a specific time period (to reduce size of captured data), while taking heap dumps into separate sessions, make it possible to do full analysis of GC memory increase/decrease tied allocation callstacks for individual object instances.
 
+### Tracing WebAssembly apps
+
+To trace a WebAssembly app:
+
+1. Add a `diagnostic_options` section to your .csproj specifying a `server` section with a `connect_url`:
+
+   ```xml
+    <ItemGroup>
+       <WasmExtraConfig Include="diagnostic_options" Value='
+    {
+        "server": {
+            "suspend": [true|false],
+            "connect_url": "ws://127.0.0.1:[port]/diagnostics"
+        }
+    }
+    ' />
+    </ItemGroup>
+   ```
+
+   Specify `"suspend":  true` to suspend the runtime at startup (**NOTE** this will make the browser unresponsive until a diagnostic tool connects and resumes it.)  The URL has to use the `ws://` scheme (`wss://` is WIP) the host should be a localhost, the port can be arbitrary, and the path `/diagnostics` is arbtrary, too.
+
+2. Run `dotnet-dsrouter server-websocket -ws http://127.0.0.1:[port]/diagnostics -ipcs /path/to/domain_socket` the `-ws` URL should be the same as the `connect_url` in the .csproj, but with the `http://` scheme.  The `-ipcs` option is some arbitrary path.
```

Comment:
nit: since its server-websocket maybe put -ipcs argument first and then -ws

btw, should we prepare to handle web socket client as well (not implement anything, just make sure we name things accordingly)? If so, maybe we should use names like -wss and potential -wsc in the future (websocket client), and maybe still use server-server but when using -wss it will use web socket and using -tcps using tcp server?

Reply:
> btw, should we prepare to handle web socket client as well (not implement anything, just make sure we name things accordingly)?

I'm not sure that a websocket client mode makes sense.  Websockets are asymmetrical - the browser starts the connections to other hosts.  There is no mechanism for a browser to listen for connections.